apps/rehash.c

```diff
@@ -328,7 +328,7 @@ static int do_dir(const char *dirname, enum Hash h)
         exit(1);
     }
     while ((filename = OPENSSL_DIR_read(&d, dirname)) != NULL) {
-        if ((copy = strdup(filename)) == NULL
+        if ((copy = OPENSSL_strdup(filename)) == NULL
```

Comment:
But in that case, the program exits...

Reply:
> But in that case, the program exits...

And some sanitizer or memory debugger will report a problem, and it will have to be fixed.. I mean if you recognize formal problem, there is no point not resolving it...

Reply:
This then becomes a cascade of freeing memory; `buf`, `files`, `copy`, and then other `exit(1)` statements above this. At this point the program is already in trouble due to lack of memory, not sure making a sanitizer happy at this point is worth it. This simple change would become quite large, and riskier.